src/components/TodoItem/Todoitem.tsx

```diff
@@ -0,0 +1,128 @@
+/* eslint-disable jsx-a11y/label-has-associated-control */
+import React, { useEffect, useRef, useState } from 'react';
+import cn from 'classnames';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  todo: Todo;
+  todosBoot: number[];
+  deleteTodo: (todoId: number) => void;
+  updateTodo: (
+    todoId: number,
+    title: string,
+    completed?: boolean,
+  ) => Promise<void> | undefined;
+};
+
+const TodoItem: React.FC<Props> = ({
+  todo,
+  todosBoot,
+  deleteTodo,
+  updateTodo,
+}) => {
+  const { id, title, completed } = todo;
+
+  const [selectTitle, setSelectTitle] = useState<string>(title);
+  const [changeTitle, setChangeTitle] = useState<boolean>(false);
+
+  useEffect(() => {
+    setSelectTitle(title);
+    setChangeTitle(false);
+  }, [title]);
+
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    if (inputRef.current) {
+      inputRef.current?.focus();
+    }
+  }, [changeTitle]);
+
+  const handleChangeTitleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    const newTitle = selectTitle.trim();
+
+    if (newTitle === title) {
+      setChangeTitle(false);
+
+      return;
+    }
+
+    if (newTitle === '') {
```

Comment:
```suggestion
    if (!newTitle) {
```
empty string is a falsy value